README.md

```diff
@@ -1,30 +1,90 @@
 
-Let's take the following Compose file. It launches an HTTP server which prints
-the specified text when hit on the configured port.
+![Image showing Docker CLI command flow](DockerAppCLIcommands.png)
+
+## Using Docker App
+
+Four primary steps comprise the Docker App process:
+1. Writing an App Definition
+1. Building an App Image
+1. Sharing the App on the Hub (optional)
+1. Running the App
+
+### Writing an App definition
+
+The first step in using Docker App is to write the App definition. This definition can be created (1) from an existing Compose file using the `docker app init` command (2) via  a template from the [Application Designer](https://docs.docker.com/ee/desktop/app-designer/), or (3) from scratch. 
+
+The App definition is a .dockerapp folder that contains three distinct pieces: metadata, a service list, and the parameters. 
+
+| File | Description |
+| :------------ | :------------ |
+| metadata.yml | metadata including the App name and version |
+| docker-compose.yml | Service list defined in a [Compose file](https://docs.docker.com/compose/compose-file/) |
+| parameters.yml | Parameters that can be changed when running the App |
+
+*Note: To store additional files in Docker Apps, such as* `prod.yml`*,* `test.yml` *or other config files, you need only to add these files to the *.dockerapp directory. All files will be packaged into the App image through the use of the* `docker app build`  *command.*
+
+### Building an App image
+
+Once the App definition is written, the next step is to build the App image from the App definition using the `docker app build` command. With this command you can tag your App image, set build-time variables, or make the build quiet. 
+Note that service images inside of an App image are immutable, meaning that the App version ties to a fixed list of service images (i.e., updating the images inside of a Docker App requires rebuilding the App image).
+
+### Sharing the App on the Hub
+
+You can push any App image already built or pulled to Docker Hub (or any registry) using the `docker app push` command. You can also pull App images from any container registry using the `docker app pull` command.
+
+When pushing an App image, all the service images used by the application are pushed at the same time inside a single entity. The version of each service image is resolved at build time from its tag.
+
+### Running the App
+
+The final Docker App step is to actually run your App using the `docker app run` command. You can either pick up an App image from Docker Hub or one that you built locally and deploy it to Swarm or Kubernetes.
+
+## Example
+
+Using the [hello-world](https://github.com/docker/app/tree/master/examples/hello-world) application example, we are going to build, share, and run a Docker App that launches an HTTP server that prints the text variable value when hit on the configured port. 
```

Comment:
You can replace https://github.com/docker/app/tree/master/examples/hello-world with `./examples/hello-world`. GitHub markdown supports relative paths.

Reply:
done